hazelcast/src/main/java/com/hazelcast/map/impl/SimpleEntryView.java

```diff
@@ -257,6 +280,10 @@ public void readData(ObjectDataInput in) throws IOException {
         // reads the deprecated evictionCriteriaNumber from the data input (client protocol compatibility)
         in.readLong();
         ttl = in.readLong();
+        //RU_COMPAT_3_10
+        if (in.getVersion().isGreaterOrEqual(V3_11)) {
```

Comment:
`SimpleEntryView` is also used when delivering WAN updates, so this change will break the implicit WAN compatibility (see also https://github.com/hazelcast/hazelcast-enterprise/issues/2231).

Reply:
It's also BinaryInterface which means these changes aren't permitted.

Reply:
Damn!

Reply:
Good catch guys. Turns out the @BinaryInterface is wrongly placed here.
Regarding the WAN compatibility, I will update https://github.com/hazelcast/hazelcast-enterprise/issues/2231 accordingly to handle this case too.